projects/plugins/protect/src/class-jetpack-protect.php

```diff
@@ -52,7 +52,19 @@ function () {
 					)
 				);
 				// Sync package.
-				$config->ensure( 'sync' );
+				$config->ensure(
+					'sync',
+					array(
+						'jetpack_sync_modules'            => array(
+							'Automattic\\Jetpack\\Sync\\Modules\\Options',
+							'Automattic\\Jetpack\\Sync\\Modules\\Callables',
+						),
+						'jetpack_sync_callable_whitelist' => array(
+							'get_plugins' => array( 'Automattic\\Jetpack\\Sync\\Functions', 'get_plugins' ),
+						),
+						'jetpack_sync_options_whitelist'  => array( 'active_plugins' ),
```

Comment:
I think you want to also set `'jetpack_sync_options_contentless' => array()`.

The idea was that an enabled module will use default filter values if custom values aren't provided. In the case of the `Options` module, there are [two associated filters](https://github.com/Automattic/jetpack/blob/master/projects/packages/sync/src/class-data-settings.php#L31). That's why the `contentless` options are shown on the sync settings debug screen when testing this PR; the default value is being used.

If you have any ideas on how to make this easier for consumers to use, that would be great!

I did some testing by activating and deactivating plugins, and checking the synced values. It seemed to work correctly in my tests.

Reply:
thanks for looking at it @kbrown9 

> I think you want to also set 'jetpack_sync_options_contentless' => array().

Addeed it and noticed the options are gone

> The idea was that an enabled module will use default filter values if custom values aren't provided.

Got it. But if values are provided they are also merged with the "must sync" defaults right. See how there are other options and callables there that are not in the list I requested.

> If you have any ideas on how to make this easier for consumers to use, that would be great!

For the most of it, it looks great. I think we only need some docs. I can draft something for your review.

One suggestion that comes to mind is that it would be great if, when adding callables, that I could do it by only using their keys, and not having to add the whole `key => callback` pair again.